osv2/mget.go

```diff
@@ -0,0 +1,146 @@
+package osv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+	"github.com/CrowdStrike/opensearchtools"
+)
+
+// MGetRequest is a marshalable form of [opensearchtools.MGetRequest] specific to the opensearchapi.MgetRequest in OpenSearch v2.
+type MGetRequest struct {
```

Comment:
Should we also have `Add` and `AddDocs` for this? So if a consumer just wants to live if OSV2 they can? Or is that a bit of overkill?

Reply:
Yes, definitely. I'll add them. Thanks.

Reply:
@cs-marshall I added them and also added a the constructor for MGetRequest to osv2/MGetRequest.
And also renamed `SetIndex` to `WithIndex` (same idea as in your PR).